scheduler/test/cook/test/mesos/mesos_mock.clj

```diff
@@ -0,0 +1,844 @@
+(ns cook.test.mesos.mesos-mock
+  (:use clojure.test)
+  (:require [clojure.core.async :as async]
+            [clojure.data.csv :as csv]
+            [clojure.java.io :as io]
+            [clojure.tools.logging :as log]
+            [cook.mesos :as c]
+            [cook.mesos.mesos-mock :as mm]
+            [cook.mesos.share :as share]
+            [cook.test.testutil :refer (restore-fresh-database! create-dummy-group create-dummy-job create-dummy-instance)]
+            [datomic.api :as d]
+            [mesomatic.scheduler :as mesos]
+            [mesomatic.types :as mesos-types])
+  (:import org.apache.curator.framework.CuratorFrameworkFactory
+           org.apache.curator.framework.state.ConnectionStateListener
+           org.apache.curator.retry.BoundedExponentialBackoffRetry))
+
+(defn string->uuid
+  "Parses the uuid if `uuid-str` is a uuid, returns nil otherwise"
+  [uuid-str]
+  (try
+    (java.util.UUID/fromString uuid-str)
+    (catch Exception e
+      nil)))
+
+(defn noop-mock-scheduler
+  []
+  (mesos/scheduler
+    (registered [this driver framework-id master-info])
+    (reregistered [this driver master-info])
+    (offer-rescinded [this driver offer-id])
+    (framework-message [this driver executor-id slave-id data])
+    (disconnected [this driver])
+    (slave-lost [this driver slave-id])
+    (executor-lost [this driver executor-id slave-id status])
+    (error [this driver message])
+    (resource-offers [this driver offers])
+    (status-update [this driver status])))
+
+(deftest make-offer
+  (let [basic-host-info {:hostname "hosta"
```

Comment:
I ended up not changing this anywhere. My reasoning is, for the tests that have the structure shown explicitly, the test will later make assertions based on the particular data in the structure. Specifically, the problem is all the functions are mutating the structure in some way so writing a generic "thing is what I expect function" would be effectively the same as the function I'm trying to test. 

The alternative would be to def the host map but in that case, a change to a value for one test will end up breaking every other test. Also, it would be less clear why the values are the way the are because they are defined elsewhere.

I'm not sure I care so much that if you feel strong about def-ing the host I will do it.